### PR TITLE
Each log entry should be on its own line

### PR DESCRIPTION
**Pull request by CoderDake**
The last line of log entries doesn't have a new line at the end so `lnav` can get confused.
Adding this small fix to address this

RELEASE_NOTE_EXCEPTION=Small fix for logging entries

### CODE REVIEW

File: packages/devtools_app/lib/src/shared/log_storage.dart [hunk 1]
```diff
@@ -45,7 +45,8 @@ class LogStorage {
             if (e.stackTrace != null) 'stackTrace': e.stackTrace.toString(),
           }),
         )
-        .join('\n');
+        .map((line) => line + '\n') // Each entry needs to be on it's own line
+        .join();
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/log_storage.dart:49]: what about adding an extension method `joinWithTrailing(Pattern pattern)` that calls join and then appends one additional pattern at the end? We have `StringExtension` in utils.dart you can add this to.

File: packages/devtools_app/lib/src/shared/log_storage.dart [hunk 2]
```diff
@@ -6,6 +6,8 @@ import 'dart:collection';
 import 'dart:convert';
 import 'package:logging/logging.dart';
 
+import '../../devtools_app.dart';
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/log_storage.dart:9]: don't import the devtools_app library - import used libraries directly

File: packages/devtools_app/lib/src/shared/primitives/utils.dart [hunk 3]
```diff
@@ -1180,6 +1180,13 @@ extension StringExtension on String {
   }
 }
 
+extension IterableExtension<T> on Iterable<T> {
+  /// Joins the iterable with [separator], and also adds a trailing [separator].
+  String joinWithTrailing([String separator = '']) {
+    return '${join(separator)}\n';
```
[the remark this answers is not in the document]
Comment (CoderDake) [reply packages/devtools_app/lib/src/shared/primitives/utils.dart:1186]: oops yep! I was moving a bit too fast on this one 😜

File: packages/devtools_app/lib/src/shared/primitives/utils.dart [hunk 4]
```diff
@@ -1180,6 +1180,13 @@ extension StringExtension on String {
   }
 }
 
+extension IterableExtension<T> on Iterable<T> {
+  /// Joins the iterable with [separator], and also adds a trailing [separator].
+  String joinWithTrailing([String separator = '']) {
+    return '${join(separator)}$separator';
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/primitives/utils.dart:1186]: on second thought, should this return the empty string if the list is empty?

**Review by kenzieschmoll — APPROVED** [review]

**Pull request merged**